packages/react/src/utility-components/AnimateHeight/AnimateHeight.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+import classes from './AnimateHeight.module.css';
+
+export type AnimateHeightProps = {
+  open: boolean;
+} & React.HTMLAttributes<HTMLDivElement>;
+
+/**
+ * AnimateHeight is a component that animates its height when the `open` prop changes.
+ */
+export const AnimateHeight = ({
+  children,
+  className,
+  open,
+  style,
+  ...rest
+}: AnimateHeightProps) => {
+  const [height, setHeight] = useState<number>(0);
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
```

Comment:
Oh, didn't catch that you changed this to `useLayoutEffect`. Not sure if we should do this because of SSR.

- https://medium.com/@alexandereardon/uselayouteffect-and-ssr-192986cdcf7a

Reply:
I changed it just because I thought the intention behind the `useLayoutEffect` hook better suits this use case. I can change back to `useCallback` if you think that's less risky.

Reply:
I changed it back now, with one additional change: I calculates the height inside the `useCallback` function (sets it to 0 if `open` is `false`). I noticed that `getBoundingClientRect` didn't work quite as expected when this was done directly as in the first commit.

Reply:
Hold on, somehow my last change wasn't pushed

Reply:
Now!

Reply:
Tested the component in the Storefront now (which is nextjs) and it seems to work fine now! 👍